src/stream/src/executor/aggregation/agg_group.rs

```diff
@@ -316,11 +317,7 @@ impl<S: StateStore, Strtg: Strategy> AggGroup<S, Strtg> {
             .expect("row count state should not be NULL")
             .as_int64();
         if row_count < 0 {
-            tracing::error!(group = ?self.group_key_row(), "bad row count");
-            if cfg!(debug_assertions) {
-                // TODO: need strict mode sys param / session var
-                panic!("row count should be non-negative");
-            }
+            inconsistency_panic!(group = ?self.group_key_row(), row_count, "row count should be non-negative");
 
             // NOTE: Here is the case that an inconsistent `DELETE` arrives at HashAgg executor, and there's no
             // corresponding group existing before (or has been deleted). In this case, `prev_row_count()` will
```

Comment:
Should we change the behavior here? according to the assumption of "trust stream" 🥵